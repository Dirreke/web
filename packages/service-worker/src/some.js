export class Client {
    test() {
        console.log("LOL IM A MODULE")
    }
}