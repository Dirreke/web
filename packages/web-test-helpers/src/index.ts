export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
