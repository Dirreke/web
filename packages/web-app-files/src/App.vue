<template>
  <main id="files" class="oc-flex oc-height-1-1">
    <div v-if="dragareaEnabled" class="dragarea" />
    <router-view tabindex="0" class="oc-width-expand" />
  </main>
</template>
<script lang="ts">
import { defineComponent, watch } from 'vue'
import { useRoute, useStore } from 'web-pkg/src/composables'
import { eventBus } from 'web-pkg/src/services/eventBus'
import { mapMutations } from 'vuex'
import { buildResource } from 'web-client/src/helpers'

export default defineComponent({
  setup() {
    const store = useStore<any>()
    watch(useRoute(), () => {
      store.dispatch('Files/resetFileSelection')
    })
  },
  data: () => ({
    dragareaEnabled: false
  }),
  created() {
    const dragOver = eventBus.subscribe('drag-over', this.onDragOver)
    const dragOut = eventBus.subscribe('drag-out', this.hideDropzone)
    const drop = eventBus.subscribe('drop', this.hideDropzone)
    const serviceWorkerEdit = eventBus.subscribe('sw.copy', this.onServiceWorkerCopy)

    this.$on('beforeDestroy', () => {
      eventBus.unsubscribe('drag-over', dragOver)
      eventBus.unsubscribe('drag-out', dragOut)
      eventBus.unsubscribe('drop', drop)
    })
  },

  methods: {
    ...mapMutations('Files', ['UPSERT_RESOURCE']),

    hideDropzone() {
      this.dragareaEnabled = false
    },
    onDragOver(event) {
      this.dragareaEnabled = (event.dataTransfer.types || []).some((e) => e === 'Files')
    },
    onServiceWorkerCopy(event) {
      const response = event.response
      const resource = buildResource(response)
      console.log(resource)
      //context.commit('UPSERT_RESOURCE', movedResource)
      //this.UPSERT_RESOURCE(resource)
      console.log("SERVICE WORKER COPY EVENT", event)
    }
  }
})
</script>

<style lang="scss" scoped>
main {
  max-height: 100%;
}

.dragarea {
  background-color: rgba(60, 130, 225, 0.21);
  pointer-events: none;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  position: absolute;
  z-index: 9;
  border-radius: 14px;
  border: 2px dashed var(--oc-color-swatch-primary-muted);
}

#files {
  position: relative;
}
</style>
