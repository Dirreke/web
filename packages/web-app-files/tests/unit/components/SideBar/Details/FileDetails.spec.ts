import FileDetails from '../../../../../src/components/SideBar/Details/FileDetails.vue'
import { ShareTypes } from 'web-client/src/helpers/share'
import { createStore, defaultComponentMocks, defaultPlugins, shallowMount } from 'web-test-helpers'
import { mockDeep } from 'jest-mock-extended'
import { SpaceResource } from 'web-client/src/helpers'
import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'

const createFile = (input) => {
  return {
    isReceivedShare: () => false,
    getDomSelector: () => input.id,
    ...input // spread input last so that input can overwrite predefined defaults
  }
}

const simpleOwnFolder = createFile({
  id: '1',
  type: 'folder',
  ownerId: 'marie',
  ownerDisplayName: 'Marie',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740'
})

const sharedFolder = createFile({
  id: '2',
  type: 'folder',
  ownerId: 'einstein',
  ownerDisplayName: 'Einstein',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740',
  shareTypes: [ShareTypes.user.value]
})

const simpleOwnFile = createFile({
  id: '3',
  type: 'file',
  ownerId: 'marie',
  ownerDisplayName: 'Marie',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  size: '740'
})

const sharedFile = createFile({
  id: '4',
  path: '/Shares/123.png',
  type: 'file',
  ownerId: 'einstein',
  ownerDisplayName: 'Einstein',
  preview: 'example.com/image',
  thumbnail: 'example.com/image',
  mdate: 'Tue, 20 Oct 2015 06:15:00 GMT',
  size: '740',
  shareTypes: [ShareTypes.user.value],
  isReceivedShare: () => true
})

describe('Details SideBar Panel', () => {
  describe('displays a resource of type folder', () => {
    describe('on a private page', () => {
      it('with timestamp, size info and (me) as owner', () => {
        const { wrapper } = createWrapper(simpleOwnFolder)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info and share date', () => {
        const { wrapper } = createWrapper(sharedFolder)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info and share date running on eos', () => {
        const { wrapper } = createWrapper(sharedFolder, false, true)
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('on a public page', () => {
      it('with owner, timestamp, size info and no share info', () => {
        const { wrapper } = createWrapper(sharedFolder, true)
        expect(wrapper).toMatchSnapshot()
      })
      it('with owner, timestamp, size info and no share info running on eos', () => {
        const { wrapper } = createWrapper(sharedFolder, true, true)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
  describe('displays a resource of type file', () => {
    describe('on a private page', () => {
      it('with timestamp, size info and (me) as owner', () => {
        const { wrapper } = createWrapper(simpleOwnFile)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info, share date and preview', () => {
        const { wrapper } = createWrapper(sharedFile)
        expect(wrapper).toMatchSnapshot()
      })
      it('with timestamp, size info, share info, share date and preview running on eos', () => {
        const { wrapper } = createWrapper(sharedFile, false, true)
        expect(wrapper).toMatchSnapshot()
      })

      it('updates when the shareTree updates', async () => {
        const { wrapper } = createWrapper(sharedFile)
        // make sure this renders once when initial sharesTree become available
        wrapper.vm.$store.state.Files.sharesTree = {
          '/Shares': [{}]
        }

        await wrapper.vm.$nextTick
        expect(wrapper).toMatchSnapshot()

        // ... and renders again when the relevant shares become available
        wrapper.vm.$store.state.Files.sharesTree = {
          '/Shares': [{}],
          '/Shares/123.png': [
            {
              shareType: 0,
              owner: {
                name: 'marie',
                displayName: 'Marie Curie'
              },
              stime: 12345
            }
          ]
        }
        await wrapper.vm.$nextTick
        expect(wrapper).toMatchSnapshot()
      })
    })
    describe('on a public page', () => {
      it('with owner, timestamp, size info, no share info and preview', () => {
        const { wrapper } = createWrapper(sharedFile, true)
        expect(wrapper).toMatchSnapshot()
      })
      it('with owner, timestamp, size info, no share info and preview running on eos', () => {
        const { wrapper } = createWrapper(sharedFile, true, true)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})

function createWrapper(testResource, publicLinkContext = false, runningOnEos = false) {
  const storeOptions = {
    ...defaultStoreMockOptions,
    getters: {
      user: function () {
        return { id: 'marie' }
      },
      configuration: function () {
        return {
          options: {
            runningOnEos
          }
        }
      }
    }
  }
  storeOptions.modules.Files.getters.highlightedFile.mockImplementation(() => testResource)
  storeOptions.modules.Files.getters.versions.mockImplementation(() => ['2'])
  storeOptions.modules.Files.getters.sharesTree.mockImplementation((state) => state.sharesTree)
  storeOptions.modules.Files.state.sharesTree = {}
  const store = createStore(storeOptions)
  return {
    wrapper: shallowMount(FileDetails, {
      global: {
        stubs: { 'router-link': true, 'oc-resource-icon': true },
        provide: {
          displayedItem: testResource,
          displayedSpace: mockDeep<SpaceResource>()
        },
        directives: {
          OcTooltip: jest.fn()
        },
        plugins: [...defaultPlugins(), store],
        mocks: {
          ...defaultComponentMocks(),
          $route: {
            meta: {
              auth: !publicLinkContext
            }
          },
          $router: jest.fn()
        }
      }
    })
  }
}
