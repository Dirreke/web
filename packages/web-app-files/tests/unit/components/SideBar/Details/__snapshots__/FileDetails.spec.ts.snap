// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details SideBar Panel displays a resource of type file on a private page updates when the shareTree updates 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <tr data-testid="shared-via">
        <th scope="col" class="oc-pr-s">Shared via</th>
        <td>
          <router-link-stub tag="a" to="[object Object]"><span>/Shares</span></router-link-stub>
        </td>
      </tr>
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page updates when the shareTree updates 2`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <tr data-testid="shared-by">
        <th scope="col" class="oc-pr-s">Shared by</th>
        <td><span>Marie Curie</span></td>
      </tr>
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page with timestamp, size info and (me) as owner 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 21, 2015, 7:28 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Marie
            <span>(me)</span>
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page with timestamp, size info, share info, share date and preview 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a private page with timestamp, size info, share info, share date and preview running on eos 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">EOS Path</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">/Shares/123.png</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy EOS path" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">Direct link</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">undefinedfiles/spaces/personal/home/Shares/123.png</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy direct link" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a public page with owner, timestamp, size info, no share info and preview 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type file on a public page with owner, timestamp, size info, no share info and preview running on eos 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div data-testid="preview" class="details-preview oc-flex oc-flex-middle oc-flex-center oc-mb" style="background-image: none;">
      <oc-spinner-stub arialabel="" size="medium"></oc-spinner-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This file has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">Oct 20, 2015, 6:15 AM</oc-button-stub>
        </td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <tr data-testid="versionsInfo">
        <th scope="col" class="oc-pr-s">Versions</th>
        <td>
          <oc-button-stub type="button" size="medium" arialabel="See all versions" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">1</oc-button-stub>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">EOS Path</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">/Shares/123.png</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy EOS path" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">Direct link</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">undefinedfiles/spaces/personal/home/Shares/123.png</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy direct link" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a private page with timestamp, size info and (me) as owner 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <!---->
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td><span>Oct 21, 2015, 7:28 AM</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Marie
            <span>(me)</span>
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <!---->
      <!---->
      <!---->
      <tr data-testid="tags">
        <th scope="col" class="oc-pr-s">Tags</th>
        <td>
          <router-link-stub tag="a" to="[object Object]"><span><span class="oc-mr-xs">moon,</span></span></router-link-stub>
          <router-link-stub tag="a" to="[object Object]"><span><span>mars</span></span></router-link-stub>
        </td>
      </tr>
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a private page with timestamp, size info, share info and share date 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This folder has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td><span>Oct 21, 2015, 7:28 AM</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <!---->
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a private page with timestamp, size info, share info and share date running on eos 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This folder has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td><span>Oct 21, 2015, 7:28 AM</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <!---->
      <tr>
        <th scope="col" class="oc-pr-s">EOS Path</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate"></p>
            <oc-button-stub type="button" size="medium" arialabel="Copy EOS path" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">Direct link</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">undefinedfiles/spaces/personal/home</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy direct link" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a public page with owner, timestamp, size info and no share info 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This folder has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td><span>Oct 21, 2015, 7:28 AM</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <!---->
      <!---->
      <!---->
      <!---->
    </table>
  </div>
</div>
`;

exports[`Details SideBar Panel displays a resource of type folder on a public page with owner, timestamp, size info and no share info running on eos 1`] = `
<div id="oc-file-details-sidebar">
  <div>
    <div class="details-icon-wrapper oc-width-1-1 oc-flex oc-flex-middle oc-flex-center oc-mb">
      <oc-resource-icon-stub resource="[object Object]" size="xxxlarge" class="details-icon"></oc-resource-icon-stub>
    </div>
    <div data-testid="sharingInfo" class="oc-flex oc-flex-middle oc-my-m">
      <oc-status-indicators-stub resource="[object Object]" indicators="[object Object]" target=""></oc-status-indicators-stub>
      <p class="oc-my-rm oc-mx-s">This folder has been shared.</p>
    </div>
    <table aria-label="Overview of the information about the selected file" class="details-table">
      <tr data-testid="timestamp">
        <th scope="col" class="oc-pr-s">Last modified</th>
        <td><span>Oct 21, 2015, 7:28 AM</span></td>
      </tr>
      <!---->
      <!---->
      <tr data-testid="ownerDisplayName">
        <th scope="col" class="oc-pr-s">Owner</th>
        <td>
          <p class="oc-m-rm">
            Einstein
            <!---->
            <!---->
          </p>
        </td>
      </tr>
      <tr data-testid="sizeInfo">
        <th scope="col" class="oc-pr-s">Size</th>
        <td>740 B</td>
      </tr>
      <!---->
      <tr>
        <th scope="col" class="oc-pr-s">EOS Path</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate"></p>
            <oc-button-stub type="button" size="medium" arialabel="Copy EOS path" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s">Direct link</th>
        <td>
          <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1">
            <p class="oc-my-rm oc-text-truncate">undefinedfiles/spaces/personal/home</p>
            <oc-button-stub type="button" size="medium" arialabel="Copy direct link" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium">
              <oc-icon-stub name="clipboard" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
            </oc-button-stub>
          </div>
        </td>
      </tr>
      <!---->
    </table>
  </div>
</div>
`;
